### ISSUE TITLE
Won't complete choices when using equals sign (--arg=<TAB>)

### ISSUE THREAD

**Issue by scottj97** [kislyuk/argcomplete#272]
Argparser allows either a space or equals sign between an option flag and its value.

With an option flag with choices, argcomplete properly completes the allowed values when there's a space between the option name and the value. But when I use an equals sign, it seems to fall back to some default, showing me files in the filesystem instead of the action's choices.

```
"""Test argcomplete.

PYTHON_ARGCOMPLETE_OK

"""

import argparse
import argcomplete

def main():
    """Run main command-line program."""
    parser = argparse.ArgumentParser(description='Play with argcomplete.')
    parser.add_argument('--verbosity', choices=['low', 'medium', 'high'])
    argcomplete.autocomplete(parser)
    args = parser.parse_args()
    print(f"{args=}")

if __name__ == "__main__":
    main()
```

When I tab complete the following:
```
./run.py --verbosity h
```
It correctly fills out the word `high`.

When I tab complete this:
```
./run.py --verbosity=h
```
I get a beep, because there's no files in cwd that start with h. (If I do `touch house`, then this will tabcomplete to `--verbosity=house`).

I've tried argcomplete 1.10.0 and also the latest master from github (b02870adab6140a306eccce03b015da134b21740).

Shouldn't argcomplete work the same whether I use a space or equals?

**Comment by kislyuk** [comment]
Thank you for your report.

Your assessment is correct - this is a bug. The reason it was not addressed before is that it's a special case in the partial argument parsing engine, and nobody happened to check that completers run when using the equals sign.

I released a fix in v1.10.1, please test and let me know if it fixes the problem for you.

**Comment by scottj97** [comment]
I confirm this fixes the issue. Thanks for the great tool and fast response!
